lfs/batcher.go

```diff
@@ -6,36 +6,43 @@ import "sync/atomic"
 // be added to the batcher from multiple goroutines and pulled off in groups
 // when one of the following conditions occurs:
 //   * The batch size is reached
+//   * Flush() is called, forcing the batch to be returned immediately, as-is
 //   * Exit() is called
-// When an Exit() occurs, the group may be smaller than the batch size.
+// When an Exit() or Flush() occurs, the group may be smaller than the batch
+// size.
 type Batcher struct {
 	exited     uint32
 	batchSize  int
 	input      chan interface{}
 	batchReady chan []interface{}
+	flush      chan interface{}
 }
 
 // NewBatcher creates a Batcher with the batchSize.
 func NewBatcher(batchSize int) *Batcher {
 	b := &Batcher{
 		batchSize:  batchSize,
-		input:      make(chan interface{}, batchSize),
+		input:      make(chan interface{}),
```

Comment:
Making input a single length channel is going to force adding of more than one item to block until `acceptInput()` pulls an item, meaning these goroutines are now forced to work in lock step. This seems a bit restrictive?